R/checks.R

```diff
@@ -598,3 +612,26 @@ check_no_tuning <- function(x) {
   rlang::abort(msg, call = NULL)
 }
 
+
+check_eval_times <- function(.times, metrics) {
+  metric_types <- tibble::as_tibble(metrics)$class
+  has_dyn <- any(metric_types == "dynamic_survival_metric")
+  if (!is.null(.times) & !has_dyn) {
+    rlang::abort("evaluation times are only used for dynamic survival metrics.")
+  }
+  if (is.null(.times) & has_dyn) {
+    rlang::abort(
+      "1+ metric require the specification of time points in the `eval_times` argument."
```

Comment:
```suggestion
      "One or more metric requires the specification of time points in the `eval_times` argument."
```

Same comment on setting `call`.